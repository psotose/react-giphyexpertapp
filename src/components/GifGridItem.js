import React from 'react'
import PropTypes from 'prop-types';

const GifGridItem = ( {title, url} ) => {  // se pone props en vez de img xq es lo que es en relidad ( props )
                                          // pero si desestructuro es mas facil de leer
  
  // console.log({id, title, url});
  return (
    <div className="card animate__animated animate__bounce">
      <img src={url} alt={title} />
      <p> { title } </p>  
      {/* { img.title } */}
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
export default GifGridItem;
