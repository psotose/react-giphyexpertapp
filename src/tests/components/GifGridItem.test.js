import React from 'react';
import { shallow } from 'enzyme'; 
import GifGridItem from '../../components/GifGridItem';

describe('Testing GifGridItem', () => {

  const title = 'Un t√≠tulo cualquiera';
  const url = 'https://algo.com/';
  const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );
  
  test('should show the component correctly', () => {  
    expect ( wrapper ).toMatchSnapshot();
  });

  test('should have a paragraph with a title', () => {
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( title );
  });

  test('should have the same img as the props url & alt', () => {
    const img = wrapper.find('img');
    // console.log( img.html() ); -> con esto veo todo el objeto, console.log( img.props() ); -> me devuelve el mismo obj xo con props en vez de atributos el
    expect( img.prop('src') ).toBe(url);
    expect( img.prop('alt') ).toBe(title);
  });

  test('should have animate__bounce class', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect( className.includes('animate__bounce') ).toBe(true);
  });
});