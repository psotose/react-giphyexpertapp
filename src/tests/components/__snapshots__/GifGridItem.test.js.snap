// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem should show the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un título cualquiera"
    src="https://algo.com/"
  />
  <p>
     
    Un título cualquiera
     
  </p>
</div>
`;
